django_ses/inbound.py

```diff
@@ -11,115 +13,129 @@
 logger = logging.getLogger(__name__)
 
 
-class UnprocessableError(Exception):
-    pass
-
-
-def parse_email(content):
-    parser = BytesParser(policy=policy.default)
-    email_message = parser.parsebytes(content)
-
-    plain_text = ''
-    html_text = ''
-    attachments = []
-
-    for part in email_message.walk():
-        content_type = part.get_content_type()
-        content_disposition = part.get('Content-Disposition', '')
-
-        # Handle plain text
-        if content_type == 'text/plain' and 'attachment' not in content_disposition:
-            plain_text = part.get_payload(decode=True).decode(part.get_content_charset() or 'utf-8')
-
-        # Handle HTML
-        elif content_type == 'text/html' and 'attachment' not in content_disposition:
-            html_text = part.get_payload(decode=True).decode(part.get_content_charset() or 'utf-8')
-
-        # Handle attachments
-        elif 'attachment' in content_disposition:
-            attachment = {
-                'filename': part.get_filename(),
-                'content_type': content_type,
-                'data': part.get_payload(decode=True)
-            }
-            attachments.append(attachment)
-
-    return plain_text, html_text, attachments
-
-
-def _get_common_headers(mail_obj):
-    common = mail_obj.get('commonHeaders', {})
-
-    return {
-        'subject': common.get('subject'),
-        'to': common.get('to'),
-        'message_id': common.get('messageId'),
-        'date': common.get('date'),
-    }
+class InboundEmail(TypedDict):
+    plain_text: str
+    html_text: str
+    attachments: array.array
+    subject: str
+    to: array.array
```

Comment:
Surprised these are using array type and not just `list`

Reply:
I'm not sure if there is any meaningful difference 🤔. I'll change them to `list`!